sdk/nodejs/resource.ts

```diff
@@ -930,6 +934,63 @@ export abstract class ProviderResource extends CustomResource {
     }
 }
 
+/**
+ * ProviderResourceReference points to an explicit provider when configuring a resource.
+ * See ProviderResource and ProviderResource.inputReference.
+ */
+export interface ProviderResourceReference {
+    /** @internal */
+    getPackage(): string;
+
+    /** @internal */
+    __registrationId?: string;
+
+    /**
+     * urn is the stable logical URN used to distinctly address a resource, both before and after deployments.
+     */
+    readonly urn: Output<URN>;
+
+    /**
+     * id is the provider-assigned unique ID for this managed resource. It is set during deployments and may be missing
+     * (undefined) during planning phases.
+     */
+    readonly id: Output<ID>;
+}
+
+export function referenceProviderResource(pkg: string, pri: Input<ProviderResource>): ProviderResourceReference {
+    let pro = ensureKnown(output(pri), "Unknown provider resource references are not supported");
```

Comment:
Unknown values don't work in provider position. Without this check the default behavior is to show this to the CLI user:

```
error: Error: failed to register new resource my-bucket-12709-ts [aws:s3/bucket:Bucket]:
2 UNKNOWN: could not parse provider reference: undefined is not a valid URN
```

We can customize the error message here to be bit more proactive on what' happening.